easybuild/tools/toolchain/toolchain.py

```diff
@@ -238,6 +238,31 @@ def is_system_toolchain(self):
         """Return boolean to indicate whether this toolchain is a system(/dummy) toolchain."""
         return is_system_toolchain(self.name)
 
+    def set_minimal_build_env(self):
+        """Set up a minimal build environment, by setting $CC, $CXX, $FC environment variables."""
+
+        # this is only relevant when using a system toolchain,
+        # for proper toolchains these variables will get set via the call to set_variables()
+        system_compilers = {
+            'gcc': 'CC',
+            'g++': 'CXX',
+        }
+
+        env_vars = {}
+        for cmd, env_var in system_compilers.items():
+            # which returns first hit from $PATH (or None if command was not found)
+            cmd_path = which(cmd)
+            if cmd_path:
+                self.log.info("Found compiler command %s at %s, so setting $%s in minimal build environment",
+                              cmd, cmd_path, env_var)
+                env_vars.update({env_var: cmd})
```

Comment:
Actually setting the environment variables is done via set `setvar` call below (here we're just collecting what to set), but yes, we don't take into account the current value for these environment variables (and that's deliberate, just like we do when using a proper toolchain).

In the case of using the system compilers, you could argue that you may want to be able to control what value is used for `$CC` and `$CXX`... Before this change you had that option (by simply defining `$CC`, since `eb` didn't), now you don't anymore.

A couple of things we could do here:

* print a clear warning when `$CC` is defined and EasyBuild is overriding that value with something different (same for `$CXX` obviously)
* provide some mechanism to be able to control the value (probably via a dedicated configure option)

Thoughts?

Reply:
Printing a warning makes a lot of sense, and it easy to implement, so done in 83e3c4a31

Reply:
I've tackled the 2nd idea too, by implementing a `--minimal-build-env` configuration option, which by default specifies to set `CC=gcc` and `CXX=g++`, but gives a whole lot more freedom now (no more hardcoding what gets set in the build environment)...

Reply:
I would have been satisfied with a warning, but this is indeed even more flexible. Nice!